site-src/guides/multiple-ns.md

```diff
@@ -1,38 +1,65 @@
+- The Gateway is configured for HTTPS and references the `foo-example-com` Secret. This allows the certificate to be managed centrally for all applications which are using this Gateway.
+- `routes.Namespaces.from: All` is the configuration to allow routes from any Namespace to use this Gateway. This impacts how routes are able to bind to this Gateway.
+
+The store team deploys their route for the `store` Service in the `store` Namespace:
+
+```yaml
+{% include 'cross-namespace-routing/store-route.yaml' %}
+```
+
+This Route has straightforward routng logic as it just matches for `/store` traffic which it sends to the `store` Service. The following snippet of the [`gateways` field](/references/spec/#networking.x-k8s.io/v1alpha1.RouteGateways) controlls which Gateways this Route can bind to:
+
+```yaml
+  gateways:
+    allow: FromList
+    gatewayRefs:
+    - name: shared-gateway
+      namespace: infra
 ```
 
-Please note that this guide illustrates this feature for HTTPRoute resource
-only as an example. The same can be accomplished with other route types as
-well.
+`gateways.allow` can be configured for Gateways in the same Namespace as the Route (the default), all Gateways, or a list of specific Gateways. In this example the store and site teams decide to reference a specific Gateway. This is the least permissive choice which ensures that other Gateways in the cluster (perhaps created in the future at some point) will not bind with these Routes. If cluster administrators have full control over how Gateways are deployed in a cluster then a more permissive binding option could be configured on Routes. The less permissive the Gateway selection is, the less that application owners need to know about which Gateways are deployed. 
+
+The site team now deploys Routes for their applications. They deploy two HTTPRoutes into the `site` Namespace:
+
+- `httproute/home` which acts as a default routing rule, matching for all traffic to `foo.example.com/*` that is not matched by an existing routing rule and routing it to `service/home`
+- `httproute/login` which routes traffic for `foo.example.com/login` to `service/login-v1` and `service/login-v2` according to the canary traffic weights 
+
+Both of these Routes use the same Gateway binding configuration which specifies `gateway/shared-gateway` in the `infra` Namespace as the only Gateway that these Routes can bind with.
+
+```yaml
+{% include 'cross-namespace-routing/site-route.yaml' %}
+```
+
+After these three Routes are deployed, they will all be bound to `gateway/shared-gateway`. The Gateway merges its bound Routes into a single flat list of routing rules. [Routing precedence](/references/spec/#networking.x-k8s.io/v1alpha1.HTTPRouteRule) within an HTTPRoute is determined by most specific match and direct conflicts between Routes would result in an error and prevent Routes from binding to that Gateway.
```

Comment:
How does the merging work? What if the both claim `prefix: /`? Or even more complex: overlapping regex

Reply:
Our [conflict resolution principles](https://gateway-api.sigs.k8s.io/concepts/guidelines/) kick in.

Reply:
Ack, thanks for the link. `More specific matches should be given precedence over less specific ones.` can be ambiguous I think (regex being the biggest issue), but that is outside the scope of this doc. I suppose it may be based on the length of the regex match (which is mostly meaningless, but best we can do).

Reply:
We should clarify regex (I thought we handled that case?)

Reply:
Is it statically and correctly possible to validate regex conflicts? I've yet to write a bug-free regex in my career so take that question from a beginners perspective.

History: We have generally stayed away from regexes because of the inherent complexity they bring.

I personally think that using with regex without another distinct metadata (host, another path prefix or header) is not going to work well for any shared infrastructure.

Reply:
any mentions of regex with routing precedence should be mentioned in the conflict resolution guide itself rather than this one so im closing this comment